File: 122 Best Time to Buy and Sell Stock II/Java/Solution.java
public class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int idx = 0;

        for (int i = 0; i < prices.length; i++) {
            int previous = prices[idx];
            int current = prices[i];

            if (current > previous) {
                max += (current - previous);
            }

            idx = i;
        }

        return max;
    }
}

File: 122 Best Time to Buy and Sell Stock II/Python/Solution.py
from typing import List, Self


class Solution:
    def maxProfit(self: Self, prices: List[int]) -> int:
        m: int = 0
        idx: int = 0
        for i in range(len(prices)):
            previous: int = prices[idx]
            current: int = prices[i]
            if current >= previous:
                m += (current - previous)
            idx = i
        return m